### DISCUSSION TITLE
`logIn()` is stuck on `/api/auth/providers` while using Credentials provider

### DISCUSSION THREAD

**Discussion by asobirov** [nextauthjs/next-auth#2588] [Help]
### Description 🐜

Next-auth logIn() is stuck endlessly on /api/auth/providers as can be seen [here.](https://i.stack.imgur.com/7kMRa.png)

`async authorize(credentials)` part doesn't seem to be executed at all, as none of the `console.log` seem to be working.

**/pages/api/auth/[...nextauth].js**
```
import Providers from 'next-auth/providers';
import connectDB from '../../../lib/connectDB';
import User from '../../../models/User';

export default NextAuth({
    //Configure JWT
    session: {
        jwt: true,
    },
    //Specify Provider
    providers: [
        Providers.Credentials({
            async authorize(credentials) {
                console.log('====THIS MESSAGE IS NOT SHOWN ON CONSOLE====')
                connectDB();
                const user = await User.findOne({
                    email: credentials.email,
                });

                console.log(user);

                if (user & (await user.matchPassword(credentials.passowrd))) {
                    client.close();
                    return { email: user.email };
                }
                client.close();
                return null;
            },
        }),
    ],
});

```
**Client** Side
```
  const onLogin = async (values) => {
        const status = await signIn('credentials', {
            redirect: false,
            email: values.email,
            password: values.password,
        });
        console.log('Login Status: ', status);
        // This is not logged as well, due to /api/auth/providers being stuck
  }
```

This question on [stackoverflow](https://stackoverflow.com/questions/68885410/next-auth-login-is-stuck-on-api-auth-providers-while-using-credentials-provid).

### Is this a bug in your own project?

Yes

### How to reproduce ☕️

Code to reproduce the issue has been provided above

### Screenshots / Logs 📽

 [Screenshot from browser console](https://i.stack.imgur.com/7kMRa.png)

No errors on next console

### Environment 🖥

  System:
    OS: macOS 12.0
    CPU: (8) arm64 Apple M1
    Memory: 152.66 MB / 8.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.4.2 - /opt/homebrew/bin/node
    Yarn: 3.0.0 - /opt/homebrew/bin/yarn
    npm: 7.20.0 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 92.0.4515.159
    Safari: 15.0

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
If `console.log('====THIS MESSAGE IS NOT SHOWN ON CONSOLE====')` is not shown, the problem happens before the `authorize()` call.

We will need more information to be able to help you. Do you have a runnable reproduction?

What is shown when you visit http://localhost:3000/api/auth/providers directly?

**Reply by asobirov** [reply]
Here is the example with the same error: https://github.com/asobirov/nextjs-chat
And http://localhost:3000/api/auth/providers is also just stuck on loading state
<img width="449" alt="Screen Shot 2021-08-23 at 16 28 39" src="https://user-images.githubusercontent.com/48186272/130440038-b1691ee9-0d2f-4e3e-851d-1572ffa85a16.png">

**Reply by balazsorban44** [reply]
I cannot see your `[...nextauth].js` file here: https://github.com/asobirov/nextjs-chat/tree/main/pages/api/auth. Make sure to add it correctly.

https://next-auth.js.org/getting-started/example

**Reply by asobirov** [reply]
Sorry, forgot to push changes. I will push them in a minute (already pushed)

**Reply by balazsorban44** [reply]
So here is the result when I visit `/api/auth/providers`:
![](https://i.imgur.com/o1y763t.png)

THe issue is definitely in your configuration somewhere. Try to open it in an incognito window to make sure it is not some kind of browser extension/addon that is causing the problem.

**Reply by asobirov** [reply]
It seems that the problem is somehow related to yarn, as starting the project with `npm run dev` does show the same result on `/api/auth/providers`.

I also noticed that I get the following message on console when using `yarn dev`, which does not appear with npm:
```lorem@ipsum nextjs-chat % yarn dev
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
event - compiled successfully
wait  - compiling...
event - compiled successfully
Watchpack Error (initial scan): TypeError: Cannot destructure property 'withFileTypes' of '(intermediate value)(intermediate value)(intermediate value)' as it is null.
Watchpack Error (initial scan): TypeError: Cannot destructure property 'withFileTypes' of '(intermediate value)(intermediate value)(intermediate value)' as it is null.
Watchpack Error (initial scan): TypeError: Cannot destructure property 'withFileTypes' of '(intermediate value)(intermediate value)(intermediate value)' as it is null.
Watchpack Error (initial scan): TypeError: Cannot destructure property 'withFileTypes' of '(intermediate value)(intermediate value)(intermediate value)' as it is null.
Watchpack Error (initial scan): TypeError: Cannot destructure property 'withFileTypes' of '(intermediate value)(intermediate value)(intermediate value)' as it is null.
```

Creating a new project with `yarn create next-app` and running it with `yarn dev` produces the same error.

**Reply by balazsorban44** [reply]
Could you share a minimal reproduction and your setup? Try running:
```sh
npx envinfo --system --binaries --browsers --npmPackages "next,next-auth,react"
```
and paste the result also here

**Reply by asobirov** [reply]
Here are the results
```
System:
    OS: macOS 12.0
    CPU: (8) arm64 Apple M1
    Memory: 82.22 MB / 8.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.4.2 - /opt/homebrew/bin/node
    Yarn: 3.0.0 - /opt/homebrew/bin/yarn
    npm: 7.20.0 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 92.0.4515.159
    Safari: 15.0
```

And minimal reproduction would probably be creating an app using `yarn create next-app`

**Reply by balazsorban44** [reply]
So there is Yarn 3 now? 👀 I am still using 1, as 2 was a pain. Could you test out and see if Yarn 1 would work? It would be nice to pinpoint the cause here.

**Reply by asobirov** [reply]
Sure, I will try it with Yarn 1. And yeah, there is v3, switch to it as I had couple of inconveniences in v2.

**Reply by balazsorban44** [reply]
I will decide to give it a shot based on your response here :sweat_smile:. If the problem is v3, then I'll probably stay on Yarn 1. (I suggest you also just remove the `node_modules` folder and re-install everything)

**Reply by asobirov** [reply]
> I suggest you also just remove the `node_modules` folder and re-install everything

Already tried it and it does not seem to help

**Reply by balazsorban44** [reply]
Could you run this https://github.com/nextauthjs/next-auth/discussions/2588#discussioncomment-1231864 inside your project folder? (I am most curious about next, next-auth, react versions)

**Reply by asobirov** [reply]
This is the output after running the command
```
System:
    OS: macOS 12.0
    CPU: (8) arm64 Apple M1
    Memory: 82.22 MB / 8.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.4.2 - /opt/homebrew/bin/node
    Yarn: 3.0.0 - /opt/homebrew/bin/yarn
    npm: 7.20.0 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 92.0.4515.159
    Safari: 15.0
```
    
   Here are the versions from package.json:
 ```
   "next": "11.1.0",
    "next-auth": "^3.28.0",
    "react": "17.0.2",
```

**Reply by asobirov** [reply]
It actually works with yarn v1, and `/api/auth/providers` does return data.
<img width="1440" alt="Screen Shot 2021-08-25 at 15 22 34" src="https://user-images.githubusercontent.com/48186272/130774369-d755f2e6-1cfb-4c0e-9639-9172b80bf820.png">

**Reply by balazsorban44** [reply]
Then I guess you will have to file a bug report at `yarn` or `next`. I am sorry.

**Reply by asobirov** [reply]
Okey, I will do that. Thanks for helping

**Reply by balazsorban44** [reply]
Feel free to tag me in your issues so I can follow along!
